org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/decompiler/DecompilerImpl.java

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+* Copyright (c) 2023 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal.decompiler;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IClassFile;
+import org.eclipse.jdt.ls.core.internal.DecompilerResult;
+import org.eclipse.jdt.ls.core.internal.IDecompiler;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+
+public abstract class DecompilerImpl implements IDecompiler {
+	private static Map<DecompilerType, Map<String, DecompilerResult>> decompilerCache = new HashMap<>();
+	private static final int CACHE_SIZE = 100;
+
+	private static Map<String, DecompilerResult> getLRUCache(int maxEntries) {
+		Map<String, DecompilerResult> map = new LinkedHashMap<>(maxEntries + 1, .75F, true) {
+			public boolean removeEldestEntry(Map.Entry eldest) {
+				return size() > maxEntries;
+			}
+		};
+		return Collections.synchronizedMap(map);
+	}
+
+	private static Map<String, DecompilerResult> getCache(DecompilerType type, int maxEntries) {
```

Comment:
I'm thinking that is it necessary to put parameter `maxEntries` here? In which case will different decompilers have different size of LRU?

Meanwhile, this private method is only used in this class and the const `CACHE_SIZE` is passed in. Looks like we can just remove this parameter and directly use `CACHE_SIZE` for simplicity.